### PR TITLE
[PEPPER-811, PEPPER-813, PEPPER-885] Initial implementation of somatic file upload.

### PR DESCRIPTION
**Pull request by otchet-broad**
## Context

[PEPPER-811](https://broadworkbench.atlassian.net/browse/PEPPER-811), [PEPPER-813](https://broadworkbench.atlassian.net/browse/PEPPER-813) and [PEPPER-885](https://broadworkbench.atlassian.net/browse/PEPPER-885) 
DSM is the entry point to share somatic results with users.  This pull request implements file upload service layer by gathering metadata and issuing signed urls so that a DSM user can upload any number of files.

Once uploaded, the files are stored in GCS and await an antivirus solution that will scan the files, disposition (delete or move) them to a final resting location, and notify DSM that the file is ready for a follow-up survey to be issued.  The antivirus cloud function will be completed in a follow-up PR.

This PR stores file metadata including how to find the uploaded file in both the relational database and Elasticsearch.

Three endpoint routes have been created to support the frontend in three ways:
1) Obtaining the signed url so the client can upload the file to GCS without DSM needing to process the bits.
2) The ability for the client to request file deletion
3) Obtaining the latest list of file uploads for a given realm/participant

Releasing requires configuration updates for test/stg/prod and needs to be documented in the release plan.  I'll seek assistance from @pegahtah so we can prepare the environment with credentials if needed.

## Checklist

- [X] I have labeled the type of changes involved using the `C-*` labels.
- [X] I have assessed potential risks and labeled using the `R-*` labels.
- [X] I have considered error handling and alerts, and added `L-*` labels as needed.
- [X] I have considered security and privacy, and added `I-*` labels as needed
- [X] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] N/A If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] N/A If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [X] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [X] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [X] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [X] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

[PEPPER-811]: https://broadworkbench.atlassian.net/browse/PEPPER-811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PEPPER-813]: https://broadworkbench.atlassian.net/browse/PEPPER-813?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PEPPER-885]: https://broadworkbench.atlassian.net/browse/PEPPER-885?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pepper-apis/config/testing-inmemorydb.conf.ctmpl
Comment (otchet-broad) [inline pepper-apis/config/testing-inmemorydb.conf.ctmpl:215]: Updated to match PEPPER-882 PR.

File: pepper-apis/dsm-core/pom.xml
Comment (otchet-broad) [inline pepper-apis/dsm-core/pom.xml:132]: Added for file name verification that Greg uses in DUOS.

File: pepper-apis/config/testing-inmemorydb.conf.ctmpl
Comment (otchet-broad) [inline pepper-apis/config/testing-inmemorydb.conf.ctmpl:110]: Heads up @pegahtah - Between PEPPER-890 and this PR, there are some changes to the fileDownload "testing" configuration.  My addition at 110 ensures the credentials object contains what DSM expects.  PEPPER-890 added what DSS expects.
When I took develop's changes after 890 merged, I got tripped up by the re-rendering.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: Why do we need both ddpParticipantId and participantId?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: Why is mimeType needed? Or is it for future reference in case we need to know the type of the file?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:72]: It looks like this becomes a legacy audit record once the file is deleted, but the queries below do not account for that?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:110]: What happens if a user deletes a file, then creates a new version of the file with the same name?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:130]: Looks like you skipped the DAO, DTO pattern?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/Dsm.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/Dsm.java:64]: Just curious, what is this file used for?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java:187]: We can take this offline,  but I am also curious about what this file does.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/somatic/result/SomaticResultMetaData.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/somatic/result/SomaticResultMetaData.java:9]: Can you add a comment saying what this class is for. (It is not yet obvious from a review perspective, but there may be a pattern I am missing.)

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java]: Nit, but this could be assigned above?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:61]: Just curious, what does 'project' refer to?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java]: Are all exceptions that can occur in this method internal errors? Likewise for `authorizeUpload`.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:184]: Seems like some of these would be 400 errors?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: How will the return of `AuthorizeResult` be processed after it is returned here?

**Review by denniscunningham — COMMENTED** [review]
Thanks for all the work and some nice code to read!
Left some questions.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: It's a future placeholder in case the file type changes.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: We need the internal ID when doing DSM related key constraints and the ddpParticipantId to communicate with DSS for PEPPER-903.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:72]: They do not.  We can let the CRCs know that a file has been uploaded and since deleted.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:110]: It saves as expected.  There's a UUID generated on the blobPath but no constraint that the "fileName" has to be unique.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:130]: I did.  I wanted to keep this with the least amount of indirection needed - since the client gets all of the information at points it didn't make sense to me to translate between.  Plenty in DSM already skips the DAO, DTO pattern.  Happy to implement if you'd like to see it.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/Dsm.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/Dsm.java:64]: Writing to Elastic.  It's the "thing" that ultimately creates the `dsm` portion of the document in participant_structured.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java]: It could.  I was going between two patterns (one in DSS and one in DSM) and ended here.  I'll update.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:61]: The GCP `project` name.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:184]: Maybe.
IMHO, the following errors are worthy of 500s.
138 - we marked the database as deleted but the GCS object wasn't.  There's nothing for a user to do - and I would expect this to alert us.
165 - DSM isn't properly configured for uploads - nothing a user can do.

The exception thrown at 144 might be worthy of a different exception and error handling - but it also might be an internal error (e.g. misconfiguration).  I assumed misconfiguration but can see where we might allow that one to proceed without error.  Have any strong feelings about this before I refactor?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: It will contain the signedUrl (and related record metadata) so the client can post the pdf into GCS and visualize the entry (or errors) in the list of uploaded files in the tab the user is working in.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/participant/ParticipantWrapper.java:187]: Routes use this "wrapper" to build a collection of data from the participants_structured index.  Happy to discuss more offline.  IMHO, this needs more testing and is the source of some of the OS1/OS2 issues.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: We also need the ddpParticipantId for updating Elastic during the `create` use case.

Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: I realize there is a pattern in place already, but I'm not a fan of storing the DDP participant ID anywhere by the participant table. It lacks key constraints, and is mirrored by a real iD. 
As I take a second look at the code, it seems like we should verify the DDP participant ID at some point in process, ideally sooner than later, and stash that and the participant ID as needed.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: What would we do differently based on the file type?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: But it also has error information?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:184]: I was mostly looking at the AuthorizeResultTypes, which are used earlier in the file.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:72]: Maybe there is nothing to do here, but I imagine for many uses of these queries the caller will have to then filter out deleted files. Is that not the case?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:130]: Mostly curious because as you mentioned there are two patterns, and I've just been dealing with code that uses *both* patterns for the same table simultaneously. I don't feel particularly strongly about the pattern, just trying to avoid the situation where both are used for the same table.
As I look at this I don't see a strong need for a DTO, but this long call signature feels like a builder might be in future revisions.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: It does. AuthorizeResult will return a response so the client can tell if the fileName meets naming requirements, if the mimeType is one in the configured list to support, or if the file exceeds the maximum size allocated.  See AuthorizeResultType.  In cases where this happens, we also send back an object that contains the settings the client should be enforcing (e.g. mime type, max file size).  In an error state, the signed URL will be null and so will the SomaticResultUpload since we're rejecting the file.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: I'm storing the `DSM` participant ID and joining to get the DDP Participant ID.  See `SQL_SELECT_DOCUMENTS_BASE`

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: Not sure.  Would you like it removed?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: Ah, so we are using the mimetype. :-)
But these bad requests will all go back with status = 500?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:72]: The caller will be able to answer the question:
What are the somatic results that have been provided to this participant.  Some may have been deleted (by users or by virus scanning) - and the study staff may be fielding phone calls related to the deleted files since a follow-up survey may have been created.
It's trivial to filter out the deleted files.

Happy to implement a change here.  What would you like to see?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: Maybe we can discuss offline? I'm still not sure why we need to store DDP participant ID here.
And it seems like we will be deep into the code before we discover a bad DDP participant ID was used in the request.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:130]: I boiled down to the same opinion.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:184]: Ah, those are handed back to the client.  It's an existing pattern DSS uses to communicate with their FileUpload pattern there.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java]: No.  Change pushed to clarify which the user can respond to and which are internal.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: FWIW, I'm using the value supplied for validation so it is memorializing information provided when the file was uploaded but otherwise doesn't really matter.  It's easy to remove.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: We are.  We validate it against the list of supported mime types and will return a 200 with a payload that indicates the mime type isn't supported.

Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: This behavior is cloned from DSS' own file upload component.

Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/somaticresults/PostSomaticResultUploadRoute.java:38]: Let's talk directly when you get a chance?

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:72]: If you feel the main use case is to return deleted entries along with live ones, then no change is needed. In any case, if you've thought about the use cases and are comfortable with this implementation, then so am I. Just wanted to be sure you understood my concern.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:49]: Thx for the DM explanation. This was my misunderstanding of the columnname annotation.
We are NOT storing the DDP participant ID in SOMATIC_DOCUMENTS_TABLE, which was my concern.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:56]: My understanding from DM is that we are using mimetype to verify file type. Although we only have one type now, this would be useful in a future where we store other types of somatic result files. 
Your call on the likelihood of that future.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:110]: I see that we don't currently offer 'by filename' query criteria, so this seems fine.
Thx for the explanation.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/SomaticResultUploadService.java:184]: Discussed in person.

**Review by denniscunningham — APPROVED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/MercuryOrderStatusListener.java
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/MercuryOrderStatusListener.java:20]: Thank you

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (pegahtah) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:291]: Are we handling none unique names with this exception as here?

**Review by pegahtah — APPROVED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java
Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:291]: We're making the files in GCS unique with the blobPath.  This is the metadata out of the file that was uploaded.  See SomaticResultUploadService line 107 for how we're building the blobPath.

Comment (otchet-broad) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java:291]: ...and I forgot to mention that there's a UNIQUE constraint on bucketName+blobPath in pepper-apis/dsm-core/src/main/resources/liquibase/PEPPER-811_somatic_documents.xml

**Pull request merged**
